File: sdk/c/src/url_shortener.c
#include "hitechcloud/url_shortener.h"
#include <stdlib.h>


/* POST /api/url-shortener/shorten - Shorten a URL */
char *hc_url_shortener_shorten(hc_http_client_t *client) {
    return hc_post(client, "/api/url-shortener/shorten", NULL);
}

/* GET /api/url-shortener/links - List links */
char *hc_url_shortener_list_links(hc_http_client_t *client) {
    return hc_get(client, "/api/url-shortener/links", NULL);
}

/* GET /api/url-shortener/config - Get shortener config */
char *hc_url_shortener_get_config(hc_http_client_t *client) {
    return hc_get(client, "/api/url-shortener/config", NULL);
}

/* GET /api/url-shortener/stats - Get link stats */
char *hc_url_shortener_get_stats(hc_http_client_t *client) {
    return hc_get(client, "/api/url-shortener/stats", NULL);
}

/* GET /api/url-shortener/links/{id} - Get link details */
char *hc_url_shortener_get_link(hc_http_client_t *client) {
    return hc_get(client, "/api/url-shortener/links/{id}", NULL);
}

/* DELETE /api/url-shortener/links/{id} - Delete link */
char *hc_url_shortener_delete_link(hc_http_client_t *client) {
    return hc_delete(client, "/api/url-shortener/links/{id}", NULL);
}

File: sdk/r/R/url_shortener.R
#' URL Shortener: shorten URLs, manage links, stats
#' @param http HttpClient object
#' @return Resource object
#' @export
UrlShortenerResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "UrlShortenerResource"
  env
}

#' POST /api/url-shortener/shorten - Shorten a URL
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
shorten <- function(self, ...) {
  self$http$post("/api/url-shortener/shorten", list(...))
}

#' GET /api/url-shortener/links - List links
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listLinks <- function(self, ...) {
  self$http$get("/api/url-shortener/links", list(...))
}

#' GET /api/url-shortener/config - Get shortener config
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getConfig <- function(self, ...) {
  self$http$get("/api/url-shortener/config", list(...))
}

#' GET /api/url-shortener/stats - Get link stats
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getStats <- function(self, ...) {
  self$http$get("/api/url-shortener/stats", list(...))
}

#' GET /api/url-shortener/links/{id} - Get link details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getLink <- function(self, ...) {
  self$http$get("/api/url-shortener/links/{id}", list(...))
}

#' DELETE /api/url-shortener/links/{id} - Delete link
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteLink <- function(self, ...) {
  self$http$delete("/api/url-shortener/links/{id}", list(...))
}

